### DISCUSSION TITLE
Authentification with Urql and Nuxt

### DISCUSSION THREAD

**Discussion by innovaweb-dev** [urql-graphql/urql#1694] [Q&A] [answered]
I build a Nuxt application and use Urql of client side for linking my app with my graphQL server.
I'm confused because I don't understand what I need to build authentification of client side.

**I found this package :**
- nuxt/auth => https://www.npmjs.com/package/@nuxtjs/auth-next
- urql/auth => https://www.npmjs.com/package/@urql/exchange-auth

**My question is :**
These packages must work together or I just need one of them to build authentification ?
Thanks

**Comment by kitten** [comment, marked as answer]
we have a docs page ok the topic of Authentication and how to use `@urql/exchange-auth`: https://formidable.com/open-source/urql/docs/advanced/authentication/

it comes down to purely API authentication and handling refresh and access tokens for GraphQL calls.

**Comment by innovaweb-dev** [comment]
Thank @kitten but I had read documentation already. 
My question was to understand how Nuxt-Auth and Urql-Auth work to implement authentification on my Nuxt App.
On the Urql-Auth examples often with React, and when I search a example with Nuxt is often with Apollo.

**Reply by JoviDeCroock** [reply]
Well that's the good part, the concept of `exchanges` is framework-agnostic so just embedding the exchange and writing the logic (auth logic is also agnostic from a framework) and you should be good to go
